testplan/web_ui/testing/src/Report/reportUtils.js

```diff
@@ -263,33 +264,34 @@ const getAssertions = (selectedEntries, displayTime) => {
             : ""
         );
       }
-      for (let i = 0; i < links.length - 1; ++i) {
+      // calculate the time elapsed between assertions
+      for (let i = links.length - 1; i > 0; --i) {
         let duration = "Unknown";
-        if (links[i].utc_time && links[i + 1].utc_time) {
-          const nextEntryTime = new Date(links[i + 1].utc_time).getTime();
+        if (links[i].utc_time && links[i - 1].utc_time) {
+          const previousEntryTime = new Date(links[i - 1].utc_time).getTime();
           const currentEntryTime = new Date(links[i].utc_time).getTime();
-          const durationMilliseconds = nextEntryTime - currentEntryTime;
+          const durationMilliseconds = currentEntryTime - previousEntryTime;
           duration = formatMilliseconds(durationMilliseconds);
         }
-        duration = "(" + duration + ")";
+        duration = "(+" + duration + ")";
```

Comment:
Is the `+` needed? Duration is always positive by definition (non-negative if precision matters and 0 is accepted after rounding/representation)

Reply:
As we discussed offline, "+" was used in the previous version of testplan and also helps the user understand that it is not the duration of the actual assertion block, but the time elapsed between them.